ncutixavier/pw-with-nodejs
# #MqXZuUk7 Build RESTFUL API with JSON Web Token

### **What about this PR?**
This PR has CRUD on the endpoints with JWT for the blog

### **Description of Task to be completed**
It has the following endpoints working:
**For User**
`POST /api/v1/users/login`

`POST /api/v1/users/signup`

**For Blog**
`GET /api/v1/blogs`

`GET /api/v1/blogs/<article_id>`

`POST /api/v1/blogs`

`PATCH /api/v1/blogs/<article_id>`

`DELETE /api/v1/blogs/<article_id>`

### **How should be manually tested?**
- After Clone the repo, run `npm install`
- run `npm run start`
- You should check the port number, and test the endpoints in postman
**N.B:** There is some endpoints that require the authorization, so you have to signup and login first in order to get access

### **Relevant pivots tracker**
[#MqXZuUk7](https://trello.com/c/MqXZuUk7/38-json-web-token-jwt), [#GZNuCwwX](https://trello.com/c/GZNuCwwX/43-delete-article), [#pOqQ7JIo](https://trello.com/c/pOqQ7JIo/48-update-article), [#93Ldvpbg](https://trello.com/c/93Ldvpbg/49-create-new-article), [#pAIyEZkr](https://trello.com/c/pAIyEZkr/41-get-article), [mee8dRIY](https://trello.com/c/mee8dRIY/47-get-all-articles)
